src/net.c

```diff
@@ -982,26 +982,20 @@ int sockgets(char *s, int *len)
         (socklist[i].handler.sock.inbuf != NULL)) {
       if (!(socklist[i].flags & SOCK_BINARY)) {
         /* look for \r too cos windows can't follow RFCs */
-        p = strchr(socklist[i].handler.sock.inbuf, '\n');
-        if (p == NULL)
-          p = strchr(socklist[i].handler.sock.inbuf, '\r');
+        p = strpbrk(socklist[i].handler.sock.inbuf, "\r\n");
         if (p != NULL) {
-          *p = 0;
+          while (*p == '\n' || *p == '\r')
+            *p++ = 0;
           if (strlen(socklist[i].handler.sock.inbuf) > 510)
             socklist[i].handler.sock.inbuf[510] = 0;
```

Comment:
Either ">= 510" or "inbuf[511]".
And why 510 when the strcpy below copies to s which is backed by a buf with sizeof 520?
And yeah I know, comments on non-changed code... but I couldn't skip it.